protocol/daemons/liquidation/client/sub_task_runner_test.go

```diff
@@ -0,0 +1,284 @@
+		setupMocks func(
+			ctx context.Context,
+			mck *mocks.QueryClient,
+			results []clobtypes.AreSubaccountsLiquidatableResponse_Result,
+		)
+		subaccountIds []satypes.SubaccountId
+
+		// expectations
+		expectedResults []clobtypes.AreSubaccountsLiquidatableResponse_Result
+		expectedError   error
+	}{
+		"Success": {
+			setupMocks: func(
+				ctx context.Context,
+				mck *mocks.QueryClient,
+				results []clobtypes.AreSubaccountsLiquidatableResponse_Result,
+			) {
+				query := &clobtypes.AreSubaccountsLiquidatableRequest{
+					SubaccountIds: []satypes.SubaccountId{
+						constants.Alice_Num0,
+						constants.Bob_Num0,
+					},
+				}
+				response := &clobtypes.AreSubaccountsLiquidatableResponse{
+					Results: results,
+				}
+				mck.On("AreSubaccountsLiquidatable", ctx, query).Return(response, nil)
+			},
+			subaccountIds: []satypes.SubaccountId{
+				constants.Alice_Num0,
+				constants.Bob_Num0,
+			},
+			expectedResults: []clobtypes.AreSubaccountsLiquidatableResponse_Result{
+				{
+					SubaccountId:   constants.Alice_Num0,
+					IsLiquidatable: true,
+				},
+				{
+					SubaccountId:   constants.Bob_Num0,
+					IsLiquidatable: false,
+				},
+			},
+		},
+		"Success - Empty": {
+			setupMocks: func(
+				ctx context.Context,
+				mck *mocks.QueryClient,
+				results []clobtypes.AreSubaccountsLiquidatableResponse_Result,
+			) {
+				query := &clobtypes.AreSubaccountsLiquidatableRequest{
+					SubaccountIds: []satypes.SubaccountId{},
+				}
+				response := &clobtypes.AreSubaccountsLiquidatableResponse{
+					Results: results,
+				}
+				mck.On("AreSubaccountsLiquidatable", ctx, query).Return(response, nil)
+			},
+			subaccountIds:   []satypes.SubaccountId{},
+			expectedResults: []clobtypes.AreSubaccountsLiquidatableResponse_Result{},
+		},
+		"Errors are propagated": {
+			setupMocks: func(
+				ctx context.Context,
+				mck *mocks.QueryClient,
+				results []clobtypes.AreSubaccountsLiquidatableResponse_Result,
+			) {
+				query := &clobtypes.AreSubaccountsLiquidatableRequest{
+					SubaccountIds: []satypes.SubaccountId{},
+				}
+				mck.On("AreSubaccountsLiquidatable", ctx, query).Return(nil, errors.New("test error"))
+			},
+			subaccountIds:   []satypes.SubaccountId{},
+			expectedResults: []clobtypes.AreSubaccountsLiquidatableResponse_Result{},
+			expectedError:   errors.New("test error"),
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			queryClientMock := &mocks.QueryClient{}
+			tc.setupMocks(grpc.Ctx, queryClientMock, tc.expectedResults)
+
+			daemon := client.NewClient(log.NewNopLogger())
+			actual, err := client.CheckCollateralizationForSubaccounts(
+				daemon,
+				grpc.Ctx,
+				queryClientMock,
+				tc.subaccountIds,
+			)
+
+			if err != nil {
+				require.EqualError(t, err, tc.expectedError.Error())
+				// The daemon initializes as unhealthy.
+				// If a request fails, the daemon will not be toggled to healthy.
+				require.Error(t, daemon.HealthCheck())
```

Comment:
same here as above - added two lines to check daemon health based on returned error.